mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -347,6 +360,24 @@ public AppendResult appendMessage(InputStream msgIn, Date internalDate, final Ma
         }
     }
 
+    public AppendResult appendMessage(Content msgIn, Date internalDate, final MailboxSession mailboxSession, boolean isRecent, Flags flagsToBeSet) throws MailboxException {
```

Comment:
that's quite a bunch of logic, is it tested in some way?

Reply:
Yes in Manager tests, refactorings were automated.